dask/dataframe/core.py

```diff
@@ -299,7 +299,7 @@ class _Frame(DaskMethodsMixin, OperatorMethodMixin):
         Values along which we partition our blocks on the index
     """
 
-    def __init__(self, dsk, name, meta, divisions):
+    def __init__(self, dsk, name, meta, divisions, lens=None):
```

Comment:
I am currently -1 to putting this in the DataFrame constructor.  I think that this is too niche of a topic I think to be on the same level as meta/divisions.  I think that we can find another way.